www/app/page5.controller.js

```diff
@@ -13,16 +13,19 @@ module.exports = function ($scope) {
   }
 
   $scope.nick = "Anon";
-  $scope.currentroom = document.domain;
+  //$scope.currentroom = document.domain;
   $scope.rooms = [];
   $scope.messages = [];
   $scope.newMessage = {};
 
   //For localhost testing
   if (document.domain == 'localhost') {
     $scope.currentroom = 'localhost:8080';
+    $scope.rooms.push({

```

Comment:
Did you check that this does not cause "localhost:8080" to be duplicated in $scope.rooms when the firebase add event is triggered ?